Sources/SPTDataLoaderResult.m

```diff
@@ -0,0 +1,91 @@
+/*
+ Copyright (c) 2015-2019 Spotify AB.
+
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ */
+#import <SPTDataLoader/SPTDataLoaderRequest.h>
+#import <SPTDataLoader/SPTDataLoaderResponse.h>
+#import <SPTDataLoader/SPTDataLoaderResult.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SPTDataLoaderResult
+
++ (instancetype)fromResponse:(SPTDataLoaderResponse *)response kind:(SPTDataLoaderResultKind)kind
+{
+    @synchronized(self.class) {
```

Comment:
there is no reason this needs to be synchronized. what shared resource is being accessed?